### DISCUSSION TITLE
Failing to load: "jinx--load-module: Jinx: jinx-mod.c not found"

### DISCUSSION THREAD

**Discussion by jonEbird** [minad/jinx#182] [Q&A]
Trying to use this module and am using the following, initial installation config:
```lisp
(use-package jinx
  :straight (jinx
             :type git
             :host github
             :repo "minad/jinx")
  :hook (emacs-startup . global-jinx-mode)
  :bind (("M-$" . jinx-correct)
         ("C-M-$" . jinx-languages)))
```
But when I evaluate that I'm greeted with the error "jinx--load-module: Jinx: jinx-mod.c not found".
I'm stepping through the `jinx--load-module` function and discovered if I set it to just `(file-name-directory (locate-library "jinx"))` then it'll get past that problem.

But then the next problem I'm running into is here:
https://github.com/minad/jinx/blob/main/jinx.el#L436-L437

Since `jinx--syntax-table` defaults to `nil`, the call fails with `(wrong-type-argument syntax-table-p nil)`

I'm running on MacOS with "GNU Emacs 29.1 (build 1, aarch64-apple-darwin23.4.0, Carbon Version 170 AppKit 2487.5)
 of 2024-04-27"

**Comment by minad** [comment]
I suspect that the module cannot be located due to the way you installed the package via Straight. Many users successfully use Jinx on MacOS, so there shouldn't be a fundamental problem.

**Reply by jonEbird** [reply]
Okay, that sounds plausible. Any suggestions on how I'd troubleshoot the installation problem?

**Comment by paskn** [comment]
You can try this solution: https://github.com/minad/jinx/discussions/160
Please note that pkg-config must be aware of enchant location. You can test it with:
`pkg-config --list-all | grep enchant`
